Core/Core/Models/Assignment.swift

```diff
@@ -209,6 +212,40 @@ extension Assignment {
             return .unlocked
         }
     }
+
+    public var submissionStatusText: String? {
+        if submission == nil || submission?.workflowState == .unsubmitted || (submission?.missing ?? false) {
+            if let dueAt = dueAt, dueAt < Clock.now {
+                return NSLocalizedString("missing", comment: "")
+            } else {
+                return nil
+            }
+        }
+
+        if submission?.late ?? false {
+            return NSLocalizedString("late", comment: "")
+        } else if let s = submission, s.type != .on_paper, s.type != .none {
```

Comment:
I think it would be better to check `submission.submittedAt != nil` here. I think there's a `not_graded` type that you would also need to check but `submittedAt` not being nil seems like a safer bet. `SubmissionStatusLabel.js` might be a good resource.